### PR TITLE
Always use the `copyToImport` logic when running SQL.

### PR DESCRIPTION
**Pull request by SanjulaGanepola**
### Changes

The setup I was running against had QCCSID 37. Also, ***SECADM** is required to run these tests so that the user profile's CCSID can be changed.

@worksofliam As we discussed this PR does not pass yet, due to the failing `checkObject` logic in `getObjectList`:

```
/home/SANJULA: /QOpenSys/usr/bin/qsh
system  "CHKOBJ OBJ(QSYS/"AÜ§#\$%Ä") OBJTYPE(*LIB) AUT(*NONE)"
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": "CPD0012:  Characters in qualifier beginning 'AÜ§#\\$%Ä) ' not valid.\nCPF0001:  Error found on *N command."
}
```

Issue is also reproduceable manually in the object browser:

image

### Checklist

* [ ] have tested my change
* [ ] have created one or more test cases
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/codefori/vscode-ibmi/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by SanjulaGanepola** [comment]
@worksofliam This initial PR was my attempt to create a test to reproduce the issues found in Issue #2059. However, in making this test, I see there is another bug with how the `checkObject` handles quotes (refer to the error message in the original issue message). You can see [here](https://github.com/codefori/vscode-ibmi/pull/2068/files#diff-5e27654e5aafb6f6b2e071aada1889501b69b005ca3eef8e8e5c096c31a50f8cR44) that for the test, I got around this by using single quotes for the entire command that `system` will call and using `pase`. In the case for `checkObject`, can this be done as well and then switched to a `pase` environment? I tried this actually, but I get back no results as you can see below. Any ideas how to proceed?

image

**Comment by worksofliam** [comment]
@SanjulaGanepola Do you mind if I/we make commits to this PR in order to attempt to fix this issue?

**Comment by worksofliam** [comment]
@SanjulaGanepola If you do `ls `/QSYS.LIB/ABCD.LIB` where `"AÜ§#$%Ä"` exists, can you share the output here?

**Comment by SanjulaGanepola** [comment]
> Do you mind if I/we make commits to this PR in order to attempt to fix this issue?

Please feel free to commit to this PR.

> If you do ls /QSYS.LIB/ABCD.LIB where "AÜ§#$%Ä" exists, can you share the output here?

image

**Comment by worksofliam** [comment]
I did kind of find part of the problem. The issue is that we were always executing ILE commands using QSH, which seems to break the encoding. If we swap out QSH with PASE, then the encoding is intact and works as expect. Now the issue is that `object_statistics` isn't working as expected. I am working with @forstie to see what he thinks.

```sql

-- set the ccsid
cl: CHGUSRPRF USRPRF(liam) CCSID(273);
select CHARACTER_CODE_SET_ID from table( QSYS2.QSYUSRINFO( USERNAME => upper('liam') ) );

-- create ugly object
cl: CRTLIB LIB("AÜ§#$%Ä") TEXT("§#$öße");
cl: CRTSRCPF FILE("AÜ§#$%Ä"/"àáãÄ£ø") RCDLEN(112) CCSID(273) TEXT("Üä$öß");
cl: ADDPFM FILE("AÜ§#$%Ä"/"àáãÄ£ø") MBR("§#$MAN") SRCTYPE(CBLLE) TEXT("§#$öße");

-- check it exists
cl: CHKOBJ OBJ(QSYS/\"AÜ§#\$%Ä\") OBJTYPE(*LIB) AUT(*NONE);

-- with sql
select 
  OBJNAME          as NAME,
  OBJTYPE          as TYPE,
  OBJATTRIBUTE     as ATTRIBUTE,
  OBJTEXT          as TEXT,
  0                as IS_SOURCE,
  IASP_NUMBER      as IASP_NUMBER,
  OBJSIZE          as SIZE,
  extract(epoch from (OBJCREATED))*1000       as CREATED,
  extract(epoch from (CHANGE_TIMESTAMP))*1000 as CHANGED,
  OBJOWNER         as OWNER,
  OBJDEFINER       as CREATED_BY
from table(QSYS2.OBJECT_STATISTICS(OBJECT_SCHEMA => '"AÜ§#$%Ä"', OBJTYPELIST => '*LIB'));
```

**Comment by forstie** [comment]
Both of these worked fine for me.
That said, the operating system has sharp edges when odd characters are used in library names.
Therefore, the user should be encouraged to avoid such names and we should focus on more important topics.

-- create ugly object
cl: CRTLIB LIB("AÜ§#$%Ä") TEXT("§#$öße");
create table  "AÜ§#$%Ä".foo(name char(10));
select * 
from table(QSYS2.OBJECT_STATISTICS(OBJECT_SCHEMA => '"AÜ§#$%Ä"', OBJTYPELIST => '*ALL'));
cl: dltlib LIB("AÜ§#$%Ä");

cl: CHGUSRPRF USRPRF(scottf) CCSID(273);
select CHARACTER_CODE_SET_ID from table( QSYS2.QSYUSRINFO( USERNAME => upper('scottf') ) );
-- create ugly object
cl: CRTLIB LIB("AÜ§#$%Ä") TEXT("§#$öße");
create table  "AÜ§#$%Ä".foo(name char(10));
select * 
from table(QSYS2.OBJECT_STATISTICS(OBJECT_SCHEMA => '"AÜ§#$%Ä"', OBJTYPELIST => '*ALL'));
cl: dltlib LIB("AÜ§#$%Ä");

**Comment by worksofliam** [comment]
@forstie can you share the system QCCSID that you ran this on?

**Comment by forstie** [comment]
65535

**Comment by SanjulaGanepola** [comment]
Updated this PR to always use the `copyToImport` logic when running SQL.

Before (Code4i):
image

Before (ACS):
image

After (Code4i):
image

**Comment by worksofliam** [comment]
@SanjulaGanepola Is it possible the test case can only pass on a system with a valid QCCSID? I am running this on a 37 and it's not looking too good:

```
code:
255
command:
'CRTLIB LIB("AÜ§#$%Ä") TEXT("§#$öße")'
signal:
null
stderr:
`CPD0020: Character '�' not valid following string '"A�       '.\nCPD0020: Character '�' not valid following string '"§#$ö�  '.\nCPF0006: Errors occurred in command.`
stdout:
''
```

File: src/testing/globalization.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    return {
+      name: `CCSID ${ccsid}`, test: async () => {
+        const connection = instance.getConnection();
+        const content = instance.getContent();
+
+        const test = testData[ccsid as keyof typeof testData];
+
+        // Temporarily update user profile default CCSID
+        const [originalUserDefaultCCSID] = await content!.ibmi.runSQL(`select CHARACTER_CODE_SET_ID from table( QSYS2.QSYUSRINFO( USERNAME => upper('${content!.ibmi.currentUser}') ) )`);
+        const newCssidRes = await connection!.runCommand({ command: `CHGUSRPRF USRPRF(${content!.ibmi.currentUser}) CCSID(${ccsid})`, noLibList: true });
```
Comment (worksofliam) [inline src/testing/globalization.ts:30]: I think we may as well take out the logic to change and check the user profile. It doesn't take affect until the user is completely disconnected and reconnected.

**Comment by SanjulaGanepola** [comment]
@worksofliam Any update on this PR?

**Comment by worksofliam** [comment]
New PR may solve this issue. I will close for now.

**Pull request closed without merging**